boot/core/src/boot/core.clj

```diff
@@ -1218,10 +1218,12 @@
   "A file filtering function factory. FIXME: more documenting here."
   [mkpred]
   (fn [criteria files & [negate?]]
-    (let [tmp?   (partial satisfies? tmpd/ITmpFile)
-          ->file #(if (tmp? %) (io/file (tmp-path %)) (io/file %))
-          pred (apply juxt (mapv mkpred criteria))]
-      (filter-files #(pred (->file %)) files negate?))))
+    (if (empty? criteria)
```

Comment:
Yep it returns `nil`.
I have been thought that you should always check using `(seq coll)` if the collection contains stuff (wonderfully in The Joy of Clojure). This has to do with how lazy sequences where implemented in Clojure and it is very different from how things are done in other Lisps (you can google `nil punning` and `nil punning next rest in Clojure`).